From e3b542b702c197bebd8fc22a259e3040134a8d53 Mon Sep 17 00:00:00 2001
From: Eli Nidam <elini@marvell.com>
Date: Tue, 12 Nov 2013 19:10:43 +0200
Subject: [PATCH 1075/1825] fix: nand: Move init SOC device multiplex Register
 from board_nand_init to SOC controler

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 5638f43b8de7900d6b9ef39938696b917412c3e3

               Initial SOC device multiplex register (bit0 & bit27) in SOC Initial
               1. bit0 - Configure NAND flush enabled
               2. Set NfArbiterEn to NAND Flash (Bootrom accidently Set NfArbiterEn to Device)
                  (alp/a375 - disable, a38x-enable)

Change-Id: I9055107ab5b419bd9ca93590e8acccd9973597e8
Signed-off-by: Eli Nidam <elini@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/4306
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c        |    7 +++++++
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvRegs.h       |    5 +++++
 2 files changed, 12 insertions(+), 0 deletions(-)
 mode change 100755 => 100644 arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
old mode 100755
new mode 100644
index 8720af1..ed35a74
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
@@ -322,6 +322,13 @@ MV_STATUS mvCtrlEnvInit(MV_VOID)
 	 */
 	MV_REG_BIT_SET(PUP_EN_REG, BIT4);
 
+	/* XXX: Following setting should be configured by u-boot */
+	MV_REG_BIT_SET(SOC_DEV_MUX_REG, BIT0); /* Configure NAND flush enabled */
+
+	/* Set NfArbiterEn to NAND Flash (Bootrom accidently Set NfArbiterEn to Device) */
+	/* Disable arbitration between device and NAND */
+	MV_REG_BIT_RESET(SOC_DEV_MUX_REG, BIT27);
+
 #if defined(MV_INCLUDE_TDM)
 	mvCtrlTdmClkCtrlConfig();
 #endif
diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvRegs.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvRegs.h
index 1da5822..b35ffd5 100644
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvRegs.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvRegs.h
@@ -87,6 +87,11 @@ extern "C" {
 #define MV_CIB_CTRL_CFG_REG                     (MV_COHERENCY_FABRIC_OFFSET + 0x80)
 
 /*
+ * SoC Device Multiplex Register
+ */
+#define SOC_DEV_MUX_REG				0x18208
+
+/*
  * Power Management Clock (PMC) Gating Control Register
  */
 #define POWER_MNG_CTRL_REG			0x18220
-- 
1.7.5.4

